### PR TITLE
Tensorflow Support 

### PR DESCRIPTION
**Pull request by Netanelshoshan**
## Summary
- Added two types of models ,LSTM and a Classifier that both inherits from BaseClassifierModel.py and BaseRegressionModel.py, without the use of a new base model. 
- Added support in freqai_interface.py and data_drawer.py (loading and saving)
- Tested them against the examples strategies and others.

## Quick changelog
- Tensorflow integration

## What's new?
Tensorflow support

## TODO:

- [x] Adding Tesnorflow support
- [x] Testing against example strategies and custom ones
- [x] Adding tests
- [ ] Building a base class for more modularity
- [ ] Adding install logic for GPU support for different platforms while running the installer
- [ ] Incremental learning support - fine tuning support. Can be ensemble of models or fine-tuning the most recent model on the latest data.

### CODE REVIEW

File: tests/freqai/test_freqai_interface.py [hunk 1]
```diff
@@ -287,6 +242,69 @@ def test_start_backtesting(mocker, freqai_conf, model, num_files, strat, caplog)
...
+@pytest.mark.parametrize('model', [
+    'LightGBMClassifier',
+    'CatboostClassifier',
+    'XGBoostClassifier',
+    'XGBoostRFClassifier',
+    'SKLearnRandomForestClassifier',
+    'PyTorchMLPClassifier',
+    'TensorFlowClassifier',
+])
+def test_extract_data_and_train_model_Classifiers(mocker, freqai_conf, model):
```
Comment (xmatthias) [inline tests/freqai/test_freqai_interface.py]: Is there a particular reason to change the sequence of these tests? (execution order in CI is randomized anyway, so the actual sequence does not matter).

at first glance, this seems to be the same test than the one we had (with additional lines) ... however moved around - which essentially makes a review more difficult (as it's no longer obvious what was actually changed).

Comment (Netanelshoshan) [reply tests/freqai/test_freqai_interface.py]: There's no particular reason. Prob got messed up while refactoring. I'll revert the chances and add the checks where necessary so we could test against it 👍🏼

Comment (Netanelshoshan) [reply tests/freqai/test_freqai_interface.py]: @xmatthias for some reason it shows that i added 152 lines to test_freqai_interface.py even though i haven't made any major changes. I only added the logic needed in  test_extract_data_and_train_model_Standard() and test_extract_data_and_train_model_Classifiers().

I reverted the changes and added the tests where necessary.

Comment (Netanelshoshan) [reply tests/freqai/test_freqai_interface.py]: I reverted the changes and added the tests where necessary.

**Comment by robcaulk** [comment]
Hello,

We already considered supporting Tensorflow, and we opted against. For details about why, you can find them in the PR:

https://github.com/freqtrade/freqtrade/pull/7807#issuecomment-1547559658

Unless something has changed regarding these points, we won't be supporting Tensorflow. We would encourage you to build your LSTM in Torch and share it with the community. 

Cheers,

Robert

**Comment by Netanelshoshan** [comment]
I understand that Rob. I'd love to contribute and work on the other features that I have suggested. Closing this one for now.

**Pull request closed without merging**